cirq-core/cirq/devices/grid_qubit.py

```diff
@@ -34,14 +34,6 @@ class _BaseGridQid(ops.Qid):
     _dimension: int
     _hash: Optional[int] = None
 
-    def __getstate__(self):
-        # Don't save hash when pickling; see #3777.
-        state = self.__dict__
-        if "_hash" in state:
-            state = state.copy()
-            del state["_hash"]
-        return state
```

Comment:
This is superseded by `__getnewargs__` and `__getnewargs_ex__`  methods implemented below. If one of those methods is present, pickling will call them to get args and then unpickling calls the class with those args (which invokes `__new__`). Otherwise, pickling would call `__getstate__` and then unpickling tries to create an instance with no args first then update it with the state; this fails if `__new__` has required args.